mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -1404,49 +1408,55 @@ impl TakerSwap {
             && self.taker_coin.is_supported_by_watchers()
             && self.maker_coin.is_supported_by_watchers()
         {
-            let taker_spends_maker_payment_fut = self.taker_coin.create_taker_spends_maker_payment_preimage(
+            let maker_payment_spend_preimage_fut = self.taker_coin.create_maker_payment_spend_preimage(
                 &self.r().maker_payment.as_ref().unwrap().tx_hex,
                 self.maker_payment_lock.load(Ordering::Relaxed) as u32,
                 self.r().other_maker_coin_htlc_pub.as_slice(),
                 &self.r().secret_hash.0,
                 &self.unique_swap_data()[..],
             );
-            let taker_refunds_payment_fut = self.taker_coin.create_taker_refunds_payment_preimage(
+
+            let time_lock = match std::env::var("USE_TEST_LOCKTIME") {
+                Ok(_) => u32::try_from(self.r().data.started_at).map_err(|err| err.to_string())?,
+                Err(_) => u32::try_from(self.r().data.taker_payment_lock).map_err(|err| err.to_string())?,
```

Comment:
How about using `SafeTypeCastingNumbers::into_or_max`?

Reply:
It was @ozkanonur's suggestion [here](https://github.com/KomodoPlatform/atomicDEX-API/pull/1552#discussion_r1030479172) to use `try_from` for the typecast, so we should include him in this discussion.

Reply:
> How about using `SafeTypeCastingNumbers::into_or_max`?

I think we shouldn't use `into_or_max` on timestamp type casts. It doesn't make sense on timestamps

Reply:
> That's even strange that `started_at` and `taker_payment_lock` are u64, but we pass everywhere as u32. 

Yeah, looks like it's passed as u32 because the field's supposed to be 32 bits in the script, but then we could declare it that way in the beginning.